### DISCUSSION TITLE
How to Remove ListBoxItem Margin

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#8631] [Q&A] [answered]
Using FluentAvalonia styles, I'm trying to get rid of the spacing between ListBox items.

Got this so far

```xaml
<Style Selector="ListBox">
    <Setter Property="BorderThickness" Value="2" />
    <Setter Property="BorderBrush" Value="{DynamicResource ControlElevationBorderBrush}" />
</Style>
<Style Selector="ListBox ListBoxItem">
    <Setter Property="MinHeight" Value="0" />
</Style>
<Style Selector="ListBox ListBoxItem /template/ ContentPresenter#PART_ContentPresenter">
    <Setter Property="Margin" Value="0" />
</Style>
```

Why does the ContentPresenter still has a margin of 2 ??

[![Screenshot-7.png](https://i.postimg.cc/mk7qNkQG/Screenshot-7.png)](https://postimg.cc/zHX0qJtd)

**Comment by maxkatz6** [comment, marked as answer]
Margin is hardcoded with a local value in the template. https://github.com/amwx/FluentAvalonia/blob/master/FluentAvalonia/Styling/BasicControls/ListBoxStyles.axaml#L76
Local value has higher priority than style setter priority.

You would need either replace template, ask to fix original style, or wait for 11.0 where values inside of the template will be reworked on the Avalonia side.

Check out this page, there are a lot of useful tips for debugging this kind of problems https://docs.avaloniaui.net/docs/styling/troubleshooting

**Reply by mysteryx93** [reply]
... great. Asked FluentAvalonia to fix it.
https://github.com/amwx/FluentAvalonia/issues/176
